core/arch/x86_code.c

```diff
@@ -53,7 +53,9 @@ get_xmm_vals(priv_mcontext_t *mc)
 #ifdef X86
     if (preserve_xmm_caller_saved()) {
         ASSERT(proc_has_feature(FEATURE_SSE));
-        if (YMM_ENABLED())
+        if (ZMM_ENABLED())
```

Comment:
It seems this needs to check the lazy cxt switching flag.  It is used on the initial thread for early injection and in other places where the lazy switch should apply.

Reply:
Done (offline). Compile time client AVX-512 code detection has been added, and the lazy switch is not initialized for the attach case if the client was AVX-512.